openemrs/spec/requests/v1/doctors_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# doctors request spec
+
+require "rails_helper"
+
+RSpec.describe "V1::DoctorsController", type: :request do
+  let(:params) { attributes_for(:user) }
+
+  describe "#create" do
+    context "when user passes in valid doctor params" do
+      it "should create a doctor with a default role" do
+        expect do
+          post v1_doctors_path, params: params
+          returned_doctor = json["data"]["doctor"]
+          expect(response).to have_http_status 201
+          expect(returned_doctor["name"]).to eq params[:name]
+          expect(returned_doctor["email"]).to eq params[:email]
+          expect(returned_doctor["date_of_birth"]).to eq params[:date_of_birth].to_s
+          expect(returned_doctor["roles"]).to include "Doctor"
+        end.to change(User, :count).by 1
+      end
+    end
+
+    context "when invalid params are passed" do
+      before do
+        params.delete :date_of_birth
+        post v1_doctors_path, params: params
+      end
+
+      it "returns an error" do
+        errors = json["errors"].first
+        expect(response).to have_http_status 422
+        expect(errors["date_of_birth"]).to include("can't be blank")
```

Comment:
What are your thoughts on "one expectation per test"? 

This is more me exploring your thoughts on this. I am not requesting any change here …yet 🙂

Reply:
Hadn't heard about them before today. Went to look it up and it feels like something that would work for the project. The only concern I initially had was something along the line of writing multiple tests to assert that each field has the expected value when a record is created but that's already taken care of with the `expect to include` pattern.

Reply:
Awesome sauce!

Reply:
I'll modify my current tests to use them. Not sure I'd want to create a new PR for it